util/refobject_base.h

```diff
@@ -51,16 +51,17 @@ namespace leveldb {
  */
 class RefObjectBase
 {
- protected:
+    // force this private so everyone is using memory fenced GetRefCount
+ private:
     volatile uint32_t m_RefCount;
 
  public:
     RefObjectBase() : m_RefCount(0) {}
     virtual ~RefObjectBase() {}
 
-    uint32_t RefInc() {return(inc_and_fetch(&m_RefCount));}
+    virtual uint32_t RefInc() {return(inc_and_fetch(&m_RefCount));}
 
-    uint32_t RefDec()
+    virtual uint32_t RefDec()
```

Comment:
I reviewed the code history.  eleveldb's RefObject used to be the base class for ErlRefObject.  It did have RefDec() as a virtual.  When the base class was unified with the "equivalent" in leveldb, no one noticed that leveldb's RefDec() was not virtual.

So yes, maybe there were side effects.  If so, they are now fixed ...